lib/transifex.js

```diff
@@ -104,4 +103,61 @@ const txResources = async function (project) {
     return slugs;
 };
 
-module.exports = {txPull, txResources};
+/**
+ * Uploads English source strings to a resource in transifex
+ * @param {string} project - project slug (for example,  "scratch-editor")
+ * @param {string} resource - resource slug (for example,  "blocks")
+ * @param {object} sourceStrings - json of source strings
+ */
+const txPush = async function (project, resource, sourceStrings) {
+    const resourceObj = {
+        data: {
+            id: `o:${ORG_NAME}:p:${project}:r:${resource}`,
+            type: 'resources'
+        }
+    };
+
+    await transifexApi.ResourceStringsAsyncUpload.upload({
+        resource: resourceObj,
+        content: JSON.stringify(sourceStrings)
+    });
+};
+
+/**
+ * Creates a new resource, and then uploads source strings to it if they are provided
+ * @param {string} project - project slug (for example,  "scratch-editor")
+ * @param {object} resource - object of resource information
+ * @param {string} resource.slug - resource slug (for example,  "blocks")
+ * @param {string} resource.name - resource name
+ * @param {string} resource.i18nType - i18n format id
+ * @param {object} resource.sourceStrings - json object of source strings
+ */
+const txCreateResource = async function (project, {slug, name, i18nType, sourceStrings}) {
+    const i18nFormat = {
+        data: {
+            id: i18nType || 'KEYVALUEJSON',
```

Comment:
I wonder if we should actually default to a type other than `KEYVALUEJSON`. I know it's what www uses,  but it's actually the worst format. There's no option for adding a description, or other metadata like character limits. We actually used the [CHROME](https://docs.transifex.com/formats/chrome-json) file format for gui etc because it at least included a description field. The newer [STRUCTURED_JSON](https://docs.transifex.com/formats/json/structured-json) provides additional fields like character limits. By not using KEYVALUEJSON as the default it would serve as a reminder when setting up new resources.

Reply:
We are at a high tier, but we haven't cared about occurrence since we got off PO files. When the same string shows up in multiple places and is extracted by gettext (po files), it tracks the source position of each occurrence. XLIFF is a standard, so that makes it easier to move from one platform to another, but we'd always have to run tools to convert from XLIFF to the format accepted by our tools. Since our tools generally accept json, it's usually easier to read and convert from/to a json format.

Reply:
Would it make sense to ticket changing the default type for now and revisit after we have braindump session 2? This seems like the kind of discussion that would benefit from having more context